src/docker-utils.coffee

```diff
@@ -97,11 +107,17 @@ do ->
 					docker.getImage(id).tagAsync({ repo, tag, force: true })
 			.catch dockerDelta.OutOfSyncError, (err) ->
 				console.log('Falling back to delta-from-empty')
-				exports.rsyncImageWithProgress(imgDest, { requestTimeout, totalTimeout, startFromEmpty: true }, onProgress)
+				exports.rsyncImageWithProgress(imgDest, { requestTimeout, totalTimeout, jwttoken, startFromEmpty: true }, onProgress)
 
 	exports.fetchImageWithProgress = (image, onProgress) ->
 		Promise.using readLockImages(), ->
-			dockerProgress.pull(image, onProgress)
+			Promise.join utils.getConfig('apiKey'), utils.getConfig('uuid'), (apiKey, uuid) ->
+				dockerOptions =
+					authconfig:
+						username: 'd_' + uuid,
+						password: apiKey,
+						serveraddress: config.registryEndpoint
```

Comment:
I'd rather we avoid using `config.registryEndpoint` as we're not using it elsewhere and it could therefore be removed. Preferably we can get the registry from the `image` as you're doing in the deltas part.